### ISSUE TITLE
Warn if a build is using a package with a bug

### ISSUE THREAD

**Issue by JohnTortugo** [dotnet/arcade#2373]
I have seen more than once a repo/branch using an old version of Arcade SDK, SignTool, other packages that have bugs that we have already fixed. This have happened for a few reasons: people hardcode versions in Versions.props, we had some values hardcoded in DefaultVersion.props, etc.

I think it would be great if we had a way to warn (fail) during a build that a package is being restored with a version that has a known bug.

@chcosta @mmitche @tmat @markwilkie

**Comment by markwilkie** [comment]
Maybe put this behind a config of some sort so repo owners can choose to fail or not....

**Comment by JohnTortugo** [comment]
What I think is: using a buggy package might lead to the build failing (best case) or the build succeed but nobody notice that behind the scenes something wrong happened.

**Comment by Chrisboh** [comment]
Do we have a way today to identify a package that would be considered buggy and should not be used?

**Comment by JohnTortugo** [comment]
I don't think so. At first I think we could modify BAR to store this information and Maestro API to be able to query it. We could store for instance, `PackageName` and `range of known buggy version`. Clients would then use the API check if the package getting restored is within any of the recorded ranges.
